apps/ui/src/queries/auction.ts

```diff
@@ -0,0 +1,88 @@
+import { useInfiniteQuery, useQuery } from '@tanstack/vue-query';
+import { MaybeRefOrGetter } from 'vue';
+import { AuctionNetworkId, getOrders } from '@/helpers/auction';
+import { AuctionDetailFragment } from '@/helpers/auction/gql/graphql';
+import { getTokenPrices } from '@/helpers/coingecko';
+import { CHAIN_IDS, COINGECKO_ASSET_PLATFORMS } from '@/helpers/constants';
+
+const LIMIT = 20;
+
+export function useBidsQuery({
+  network,
+  auction
+}: {
+  network: MaybeRefOrGetter<AuctionNetworkId>;
+  auction: MaybeRefOrGetter<AuctionDetailFragment>;
+}) {
+  return useInfiniteQuery({
+    initialPageParam: 0,
+    queryKey: ['auction', network, () => toValue(auction).id, 'orders'],
+    queryFn: ({ pageParam }) =>
+      getOrders(toValue(auction).id, toValue(network), {
+        first: LIMIT,
+        skip: pageParam
+      }),
+    getNextPageParam: (lastPage, pages) => {
+      if (lastPage.length < LIMIT) return null;
+
+      return pages.length * LIMIT;
+    }
+  });
+}
+
+export function useRecentBidsQuery({
+  network,
+  auction
+}: {
+  network: MaybeRefOrGetter<AuctionNetworkId>;
+  auction: MaybeRefOrGetter<AuctionDetailFragment>;
+}) {
+  return useQuery({
+    queryKey: ['auction', network, () => toValue(auction).id, 'recentOrders'],
+    queryFn: () =>
+      getOrders(toValue(auction).id, toValue(network), {
+        first: 5,
+        orderBy: 'timestamp',
+        orderDirection: 'desc'
+      })
+  });
+}
+
+export function useBiddingTokenPriceQuery({
+  network,
+  auction
+}: {
+  network: MaybeRefOrGetter<AuctionNetworkId>;
+  auction: MaybeRefOrGetter<AuctionDetailFragment>;
+}) {
+  return useQuery({
+    queryKey: [
+      'auction',
+      network,
+      () => toValue(auction).id,
+      'biddingTokenPrice'
+    ],
+    queryFn: async () => {
+      const networkValue = toValue(network);
+      const auctionValue = toValue(auction);
+
+      const chainId = CHAIN_IDS[networkValue];
+      if (!(chainId in COINGECKO_ASSET_PLATFORMS)) {
+        return 0;
+      }
+
+      const coingeckoAssetPlatform =
+        COINGECKO_ASSET_PLATFORMS[
+          chainId as keyof typeof COINGECKO_ASSET_PLATFORMS
+        ];
+
+      const coins = await getTokenPrices(coingeckoAssetPlatform, [
+        auctionValue.addressBiddingToken
+      ]);
+
+      return (
+        coins[auctionValue.addressBiddingToken.toLocaleLowerCase()]?.usd ?? 0
```

Comment:
Why not just the regular `toLowerCase()` ?

Reply:
I think in this case we actually want to do `toLowerCase`. Will update it. There might be some edge cases in some locales that make it work in unexpected keys.

Also I will take a look at our usage in the app to find potentially broken code.